pvlib/tools.py

```diff
@@ -251,3 +252,99 @@ def _build_kwargs(keys, input_dict):
             pass
 
     return kwargs
+
+
+def _array_newton(func, x0, fprime, args, tol, maxiter, fprime2,
```

Comment:
A comment that links to scipy and discusses context would be helpful here.

Reply:
✔️  I added a comment and a FIXME